### PR TITLE
feat: Profiles + initial architecture 

### PR DESCRIPTION
**Pull request by napalmpapalam**

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 
 echo "Building the canisters..."
-cargo build --workspace --release --target wasm32-unknown-unknown --exclude common
+cargo build --workspace --release --target wasm32-unknown-unknown --exclude common --exclude tests
 
 mkdir -p wasm
 
 echo "Compressing wasm files..."
-gzip -r `find target/wasm32-unknown-unknown/release -type f -name "*.wasm"`
+gzip -r `find target/wasm32-unknown-unknown/release -type f -name "*.wasm"` -f
```
Comment (rem-code-s) [inline scripts/build.sh:12]: is a `force` needed?

File: scripts/local_create.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+dfx start --clean --background
+dfx canister create index_profile --identity catalyze_development --with-cycles 10000000000000 --no-wallet
```
Comment (rem-code-s) [inline scripts/local_create.sh]: you can also use `dfx deploy index_profile --network development  --identity catalyze_development --with-cycles 10000000000000 --no-wallet '(vec {principal "syzio-xu6ca-burmx-4afo2-ojpcw-e75j3-m67o5-s5bes-5vvsv-du3t4-wae"})'` which creates and installs the canister with 1 command

File: src/index_profile/src/calls.rs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+fn is_proxy_guard() -> Result<(), String> {
+    if is_developer().is_ok() {
+        return Ok(());
+    }
+
+    is_proxy(Proxies::default().get().expect("Failed to get proxies"))
+}
+
+#[init]
+async fn init(proxies: Vec<Principal>) {
```
Comment (rem-code-s) [inline src/index_profile/src/calls.rs]: This doenst have to be async

**Review by rem-code-s — APPROVED** [review]

File: scripts/build.sh (hunk 1 above)
Comment (napalmpapalam) [reply scripts/build.sh:12]: I dont remember for what, but it was needed :)

**Pull request merged**